File: backend/src/api/v1/categories.py
"""Category API endpoints."""
from typing import List
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from src.config.database import get_db
from src.repositories.category_repository import CategoryRepository
from src.schemas.category_schemas import CategoryResponse
from src.utils.logger import get_logger


logger = get_logger(__name__)
router = APIRouter(prefix="/categories", tags=["Categories"])


def get_category_repository(db: Session = Depends(get_db)) -> CategoryRepository:
    """Dependency for category repository."""
    return CategoryRepository(db)


@router.get("", response_model=List[CategoryResponse])
async def get_categories(
    include_inactive: bool = False,
    repository: CategoryRepository = Depends(get_category_repository)
):
    """
    Get all categories.
    
    Args:
        include_inactive: Whether to include inactive categories
        repository: Category repository
        
    Returns:
        List of categories sorted by display order
    """
    try:
        if include_inactive:
            categories = repository.get_all_categories(include_inactive=True)
        else:
            categories = repository.get_active_categories()
        
        logger.info(f"Retrieved {len(categories)} categories (include_inactive={include_inactive})")
        return categories
    
    except Exception as e:
        logger.error(f"Failed to get categories: {str(e)}", exc_info=True)
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail="Failed to retrieve categories"
        )

File: frontend/src/api/categories.ts
/**
 * API service for categories endpoints
 */
import apiClient from './client';

export interface Category {
  id: string;
  name: string;
  description: string | null;
  display_order: number;
  is_active: boolean;
}

/**
 * Get active categories
 */
export const getCategories = async (): Promise<Category[]> => {
  const response = await apiClient.get<Category[]>('/api/v1/categories');
  return response.data;
};

/**
 * Get all categories (including inactive) - admin only
 */
export const getAllCategories = async (includeInactive: boolean = false): Promise<Category[]> => {
  const response = await apiClient.get<Category[]>('/api/v1/categories', {
    params: { include_inactive: includeInactive }
  });
  return response.data;
};
